pkg/version/version_test.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package version
+
+import (
+	"reflect"
+	"testing"
+)
+
+func TestGetMinimumProviderVersions(t *testing.T) {
+	cases := []struct {
+		desc                string
+		minProviderVersions string
+		expectedMap         map[string]string
+	}{
+		{
+			desc:                "no min provider version provided",
+			minProviderVersions: "",
+			expectedMap:         make(map[string]string),
+		},
+		{
+			desc:                "single min provider version provided",
+			minProviderVersions: "provider1=0.0.2",
+			expectedMap:         map[string]string{"provider1": "0.0.2"},
+		},
+		{
+			desc:                "more than one provider version provided",
+			minProviderVersions: "provider1=0.0.2,provider2=0.0.4",
+			expectedMap:         map[string]string{"provider1": "0.0.2", "provider2": "0.0.4"},
+		},
+		{
+			desc:                "white space in min provider versions",
+			minProviderVersions: "provider1=0.0.2, provider2=0.0.4",
+			expectedMap:         map[string]string{"provider1": "0.0.2", "provider2": "0.0.4"},
+		},
+		{
+			desc:                "more white space in min provider versions",
+			minProviderVersions: "provider1=0.0.2 , provider2=0.0.4",
+			expectedMap:         map[string]string{"provider1": "0.0.2", "provider2": "0.0.4"},
+		},
```

Comment:
maybe also add few more tests for invalid semver and test for ',' and '='

Reply:
done!